src/native/fs/fs_napi.cpp

```diff
@@ -158,6 +202,110 @@ set_fs_worker_stats(Napi::Env env, Napi::Object fs_worker_stats, std::string wor
     fs_worker_stats["error"] = Napi::Number::New(env, error);
 }
 
+static int 
+get_single_user_xattr(int fd, std::string key, std::string &value) {
+    #ifdef __APPLE__
+        ssize_t value_len = fgetxattr(fd, key.c_str(), NULL, 0, 0, 0);
+    #else
+        ssize_t value_len = fgetxattr(fd, key.c_str(), NULL, 0);
+    #endif
+
+    if (value_len == -1) {
+        return -1;
+    }
+
+    if (value_len > 0) {
+        Buf val(value_len);
+
+        #ifdef __APPLE__
+            value_len = fgetxattr(fd, key.c_str(), val.cdata(), value_len, 0, 0);
+        #else
+            value_len = fgetxattr(fd, key.c_str(), val.cdata(), value_len);
+        #endif
+
+        if (value_len == -1) {
+            return -1;
+        }
+        
+        value = std::string(val.cdata(), value_len);
+    } else if (value_len == 0) {
+        value = "";
+    }
+    return 0;
+}
+
+static int 
+get_fd_xattr(int fd, XattrMap &xattr, bool skip_user_xattr)
+{
+    if (skip_user_xattr) { // we won't list the attributes just return the prefefined list
+        for (std::string key: USER_XATTRS) {
```

Comment:
I think you can iterate with a const ref instead like so - 
https://stackoverflow.com/questions/15176104/range-based-loop-get-item-by-value-or-reference-to-const
```suggestion
        for (auto const &key: USER_XATTRS) {
```